File: web-proyecto-integrador/src/types/areaPuestoCompatible.ts
export interface Area {
  id: number;
  nombre: string;
}

export interface Puesto {
  id: number;
  nombre: string;
}

export interface AreaPuestoCompatible {
  id: number;
  area: Area;
  puesto: Puesto;
}
export interface PuestoConAreas {
  id: number;
  nombre: string;
  areaIds: number[];
}

File: api-proyecto-integrador/schemas/area_puesto_compatible.py
from pydantic import BaseModel

class AreaOut(BaseModel):
    id: int
    nombre: str

    model_config = {"from_attributes": True}


class PuestoOut(BaseModel):
    id: int
    nombre: str

    model_config = {"from_attributes": True}


class AreaPuestoCompatibleOut(BaseModel):
    id: int
    area: AreaOut
    puesto: PuestoOut

    model_config = {"from_attributes": True}